lib/fuser_basic.h

```diff
@@ -56,7 +56,7 @@ class BasicGateFuser final : public Fuser<IO, Gate> {
    * two-qubit gates will get fused. To respect specific time boundaries while
    * fusing gates, use the other version of this method below.
    * @param param Options for gate fusion.
-   * @param num_qubits The number of qubits acted on by 'gates'.
+   * @param max_qubit1 The maximum qubit index (plus one) acted on by 'gates'.
```

Comment:
I assume this change is to more accurately describe situations where e.g. qubits {1, 2, 4} are being acted on? Would providing the exact set of qubits acted on be worthwhile, to ensure that a "skipped" qubit (3 in the previous example) isn't being erroneously acted on?

Reply:
This change is to describe situations where e.g. qubits {1, 2, 4} and max_qubit1 (previously num_qubits) is smaller than 4+1. max_qubit1 is used to allocate an array in line 165 and there will be out of bound array access if any qubit acted on by gates is greater or equal to max_qubit1. It is OK if there are "skipped" qubits.